### PR TITLE
mosh: support cross compile

### PR DESCRIPTION
**Pull request by squalus**
###### Description of changes

Adds support for cross compiling mosh. Tested the cross compile from amd64 and tested execution on aarch64.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
adding `strictDeps = true` on such PRs helps preventing unintentional regressions in the future.

**Comment by squalus** [comment]
Added `strictDeps = true` and tested native and cross compilation

File: pkgs/tools/networking/mosh/default.nix [hunk 1]
```diff
@@ -11,11 +11,13 @@ stdenv.mkDerivation rec {
     sha256 = "05hjhlp6lk8yjcy59zywpf0r6s0h0b9zxq0lw66dh9x8vxrhaq6s";
   };
 
-  nativeBuildInputs = [ autoreconfHook pkg-config makeWrapper ];
+  nativeBuildInputs = [ autoreconfHook pkg-config makeWrapper protobuf ]
+    ++ (with perlPackages; [ perl IOTty ]);
   buildInputs = [ protobuf ncurses zlib openssl bash-completion ]
```
Comment (Mindavi) [inline pkgs/tools/networking/mosh/default.nix:16]: I guess protobuf is still needed in buildInput, but did you check by chance?

**Review by Mindavi — APPROVED** [review]
LGTM

File: pkgs/tools/networking/mosh/default.nix (hunk 1 above)
Comment (squalus) [reply pkgs/tools/networking/mosh/default.nix:16]: Just checked, and it is needed. The binaries dynamically link to libprotobuf.

Comment (Mindavi) [reply pkgs/tools/networking/mosh/default.nix:16]: Yeah, that's what I thought. Thanks for checking 🙂

**Comment by squalus** [comment]
Oops, there's a problem with this - `bin/.mosh-wrapped` now has an unpatched perl shebang line.

File: pkgs/tools/networking/mosh/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/tools/networking/mosh/default.nix:16]: i use diffoscope to compare the result when turning strictDeps on

**Comment by Artturin** [comment]
> Oops, there's a problem with this - `bin/.mosh-wrapped` now has an unpatched perl shebang line.

perl has to be added to buildInputs too

**Comment by squalus** [comment]
Added perl back to the buildInputs, as it's a runtime dependency.

Removed the dependency on `perlPackages.IOTty`. That package doesn't cross compile. It's not necessary and works fine without it. Upstream removed it from their debian control file in 1.2.4. Link: https://github.com/mobile-shell/mosh/commit/8290e387af43ed0ed07e3ff9ae338ff624692ae3. The current control file does not depend on it either: https://github.com/mobile-shell/mosh/blob/master/debian/control

**Pull request merged**